tests/jest/rooms.js

```diff
@@ -0,0 +1,209 @@
+
+    await page.evaluate(
+      async (bob, roomId) => {
+        const alice = await aliceChatManager.connect()
+        alice.subscribeToRoomMultipart({
+          roomId: roomId,
+          hooks: {
+            onUserJoined: newUser =>
+              (window.actual = {
+                user: newUser,
+              }),
+          },
+        })
+
+        await alice.addUserToRoom({
+          userId: bob.id,
+          roomId: roomId,
+        })
+      },
+      bob,
+      room.id,
+    )
+
+    await helpers.withHook(actual => {
+      expect(actual.user.id).toBe(bob.id)
+    })
+  })
+
+  test("can have users removed", async () => {
+    const room = await helpers.makeRoom({ members: [alice, bob] })
+
+    await page.evaluate(
+      async (bob, roomId) => {
+        const alice = await aliceChatManager.connect()
+        alice.subscribeToRoomMultipart({
+          roomId: roomId,
+          hooks: {
+            onUserLeft: user =>
+              (window.actual = {
+                user: user,
+              }),
+          },
+        })
+
+        await alice.removeUserFromRoom({
+          userId: bob.id,
+          roomId: roomId,
+        })
+      },
+      bob,
+      room.id,
+    )
+
+    await helpers.withHook(actual => {
+      expect(actual.user.id).toBe(bob.id)
+    })
+  })
+
+  test("can be joined with correct permission", async () => {
+    const room = await helpers.makeRoom({ members: [alice] })
+
+    await page.evaluate(async roomId => {
+      const alice = await aliceChatManager.connect()
+      await alice.subscribeToRoomMultipart({
+        roomId: roomId,
+        hooks: {
+          onUserJoined: user =>
+            (window.actual = {
+              user: user,
+            }),
+        },
+      })
+
+      const bob = await bobChatManager.connect()
+      await bob.joinRoom({
+        roomId: roomId,
+      })
+    }, room.id)
+
+    await helpers.withHook(actual => {
+      expect(actual.user.id).toBe(bob.id)
+    })
+  })
+
+  test("can be updated", async () => {
+    const room = await helpers.makeRoom({ members: [alice] })
+
+    const updatedName = "newName"
+    const updatedData = "bar"
+    expect(room.name).not.toBe(updatedName)
+    expect(room.customData).toBeUndefined
+    expect(room.private).toBeFalsy
+
+    await page.evaluate(
+      async (roomId, updatedName, updatedData) => {
+        const alice = await aliceChatManager.connect({
+          onRoomUpdated: room => (window.actual = { room: room }),
+        })
+        await alice.updateRoom({
+          roomId: roomId,
+          name: updatedName,
+          customData: { foo: updatedData },
+          private: false,
+        })
+      },
+      room.id,
+      updatedName,
+      updatedData,
+    )
+
+    await helpers.withHook(actual => {
+      expect(actual.room.name).toBe(updatedName)
+      expect(actual.room.customData.foo).toBe(updatedData)
+      expect(actual.room.private).toBeTruthy
```

Comment:
```suggestion
      expect(actual.room.isPrivate).toBeTruthy()
```

and also, did you mean to set `private: true` above?

Reply:
maybe